### PR TITLE
doc/release-notes: update luminous notes

### PR DESCRIPTION
**Pull request by liewegas**
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: doc/release-notes.rst [hunk 1]
```diff
@@ -5,11 +5,272 @@
...
+  * RGW now supports *dynamic bucket index sharding*.  As the number
+    of objects in a bucket grows, RGW will automatically reshard the
+    bucket index in response.  No user intervention or bucket size
+    capacity planning is required.
+  * RGW has consolidated the several metadata index pools via the use of rados
+    namespaces.
+
+- *RBD*:
+
+  * RBD now has full, stable support for *erasure coded pools* via the new
```
Comment (dillaman) [inline doc/release-notes.rst]: Nit: this was added in Kraken -- so this should probably move down to the "major changes from Jewel" section with an added caveat that it requires Bluestore and EC-overwrites enabled on the data pool.

File: doc/release-notes.rst [hunk 2]
```diff
@@ -5,11 +5,272 @@
...
+    serviced via RGW itself and now supports custom metadata fields
+  * RGW now supports *dynamic bucket index sharding*.  As the number
+    of objects in a bucket grows, RGW will automatically reshard the
+    bucket index in response.  No user intervention or bucket size
+    capacity planning is required.
+  * RGW has consolidated the several metadata index pools via the use of rados
+    namespaces.
+
+- *RBD*:
+
```
Comment (dillaman) [inline doc/release-notes.rst]: * RBD mirroring's *rbd-mirror* daemon is now highly available.  We recommend deploying several instances of *rbd-mirror* for reliability.

File: doc/release-notes.rst [hunk 3]
```diff
@@ -5,11 +5,272 @@
...
+    period for failures where the host remains up but the ceph-osd
+    process is no longer running.
+  * The size of encoded OSDMaps has been reduced.
+  * The OSDs now quiesce scrubbing when recovery or rebalancing is in progress.
+
+- *RGW*:
+
+  * RGW now supports the S3 multipart object copy-part API.
+  * It is possible now to reshard an existing bucket. Note that bucket
+    resharding currently requires that all IO (especially writes) to
```
Comment (theanalyst) [inline doc/release-notes.rst]: probably conflicts with dynamic resharding we mention in the earlier section... cc @oritwas

File: doc/release-notes.rst [hunk 4]
```diff
@@ -5,11 +5,272 @@
...
+    of active MDS servers may be adjusted up or down on an active CephFS file
+    system.
+  * CephFS *directory fragmentation* is now stable and enabled by
+    default on new filesystems.  To enable it on existing filesystems
+    use "ceph fs set <fs_name> allow_dirfrags".  Large or very busy
+    directories are sharded and (potentially) distributed across
+    multiple MDS daemons automatically.
+  * Directory subtrees can be explicitly pinned to specific MDS daemons in
+    cases where the automatic load balancing is not desired or effective.
+
```
Comment (theanalyst) [inline doc/release-notes.rst]: add another note on ceph cli  `ceph status` change, probably under a tools section?

File: doc/release-notes.rst (hunk 1 above)
Comment (liewegas) [inline doc/release-notes.rst]: It wasn't stable in kraken, so I moved it up here (same for the rados feature).

File: doc/release-notes.rst (hunk 3 above)
Comment (liewegas) [inline doc/release-notes.rst]: clarified (i think?)

**Comment by theanalyst** [comment]
@liewegas could you also add the stuff from https://github.com/ceph/ceph/pull/15640 into this notably RGW policy work

File: doc/release-notes.rst [hunk 5]
```diff
@@ -5,11 +5,276 @@
...
+- *RGW*:
+
+  * RGW introduces *server side encryption* of uploaded objects with
+    three options for the management of encryption keys: automatic
+    encryption (only recommended for test setups), customer provided
+    keys similar to Amazon SSE-C specification, and through the use of
+    an external key management service (Openstack Barbician) similar
+    to Amazon SSE-KMS specification.
+  * RGW *metadata search* with ElasticSearch now supports end user requests
+    serviced via RGW itself and now supports custom metadata fields
```
Comment (yehudasa) [inline doc/release-notes.rst]: RGW metadata search backed by ElasticSearch now supports end user requests service via RGW itself, and also supports custom metadata fields. A query language was defined and a set of RESTful APIs were created for users to be able to search objects by their metadata. New APIs that allow control of custom metadata fields were also added.

File: doc/release-notes.rst [hunk 6]
```diff
@@ -5,11 +5,276 @@
...
+      intervening file system like XFS.  This provides greater performance
+      and features. FIXME DOCS
+    - BlueStore supports *full data and metadata checksums* of all
+      data stored by Ceph.
+    - BlueStore supports inline compression using zlib, snappy, or LZ4.  (Ceph
+      also supports zstd for RGW compression but zstd is not recommended for
+      BlueStore for performance reasons.)  FIXME DOCS
+
+  * *Erasure coded* pools now have full support for *overwrites*,
+    allowing them to be used with RBD and CephFS.
```
Comment (jdurgin) [inline doc/release-notes.rst]: if we're linking to docs, http://docs.ceph.com/docs/master/rados/operations/erasure-code/#erasure-coding-with-overwrites

File: doc/release-notes.rst [hunk 7]
```diff
@@ -5,11 +5,276 @@
...
+
```
Comment (jdurgin) [inline doc/release-notes.rst]: all the osd sleep settings now work. osd_recovery_sleep is probably the most important to call out

File: doc/release-notes.rst (hunk 5 above)
Comment (mattbenjamin) [inline doc/release-notes.rst]: bucket policy?

File: doc/release-notes.rst [hunk 8]
```diff
@@ -5,11 +5,387 @@
...
+      refuses IO, when the cluster warns about being close to full,
+      when an OSD will defer rebalancing a PG to itself,
+      respectively).
+    - ``ceph osd reweightn`` will specify the `reweight` values for
+      multiple OSDs in a single command.  This is equivalent to a series of
+      ``ceph osd reweight`` commands.
+    - ``ceph crush class {create,rm,ls}`` manage the new CRUSH *device
+      class* feature.  ``ceph crush set-device-class <osd> <class>``
+      will set the clas for a particular device.
+    - ``ceph mon feature ls`` will list monitor features recorded in the
```
Comment (Liuchang0812) [inline doc/release-notes.rst]: Hi, @liewegas , by any chance, should we introduce `ceph tell x help` command here? it is useful to user.

**Pull request merged**